File: Braintree/src/main/java/com/braintreepayments/api/exceptions/BraintreeException.java
package com.braintreepayments.api.exceptions;

import java.io.IOException;

/**
 * Parent class for exceptions encountered when using the SDK.
 */
public class BraintreeException extends IOException {

    public BraintreeException(String message, Throwable cause) {
        super(message, cause);
    }

    public BraintreeException(String message) {
        super(message);
    }

    public BraintreeException() {
        super();
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeException.kt
package com.braintreepayments.api

import androidx.annotation.RestrictTo
import java.io.IOException

/**
 * Parent class for exceptions encountered when using the SDK.
 */
open class BraintreeException @JvmOverloads @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP) constructor(
    message: String? = null,
    cause: Throwable? = null
) : IOException(message, cause)
